### PR TITLE
Migrate to Python 3

### PR DESCRIPTION
**Pull request by AleksandrIakhnev**
Bulk support is temporarily neglected  because it is already broken (No MMfile dependency), should be fixed.
ExecuteQuery.py formatted according to  PEP8

### CODE REVIEW

File: http_calls.py [hunk 1]
```diff
@@ -139,305 +130,284 @@ def get_attr_by_type(self,data, attr):
...
-                    k = int(k)
-                elif isinstance(val, dict) and k.isdigit():
-                    k = val.keys()[int(k)]
-                val = val[ k ]
-        except BaseException, e:
-            print 'Exception: ', e
-            val = ''
-        return val
-
-    def getRawRequestResult( self ):
```
Comment (AntonTimiskov) [inline http_calls.py:269]: Why it was removed?
It's a pretty feature to check raw responses. Used in DMP.

File: http_calls.py [hunk 2]
```diff
@@ -139,305 +130,284 @@ def get_attr_by_type(self,data, attr):
...
         return lastResponseTime
 
-    def getAttribute( self, AttrName ):
-
-        global lastRequestResult
-
-        val = json.loads( lastRequestResult )
-        try:          
-            for k in AttrName.split( '.' ):
-                if isinstance(val, list) and k.isdigit():
```
Comment (AntonTimiskov) [inline http_calls.py:259]: It was for:
`{ hits: [ { id: hit1, name: name1 } ] }`
to query `hits.0.name`

File: http_calls.py [hunk 3]
```diff
@@ -1,132 +1,123 @@
...
 
-        return self.request(self.get_full_url(url), data, headers)
-
-    def get_str(self, url, args = None):
-
-        return self.GET(self.get_full_url(url), {'Accept':'application/json'}, args)
+    def get_str(self, url, args=None):
+        return self.GET(self.get_full_url(url), {'Accept': 'application/json'}, args).decode('utf-8')
 
-    def get_hex_str( self, url, args = None ):
```
Comment (AntonTimiskov) [inline http_calls.py:102]: please revert

File: http_calls.py [hunk 4]
```diff
@@ -139,305 +130,284 @@ def get_attr_by_type(self,data, attr):
...
     def wait(self, wait_sec, retries, func, **kwargs):
 
         for r in range(int(retries)):
             if func(kwargs):
                 return True
-            self.sleep(wait_sec)        
+            self.sleep(wait_sec)
         return False
 
-    def quoteUrl( self, url ):
```
Comment (AntonTimiskov) [inline http_calls.py:169]: please revert

File: http_calls.py (hunk 2 above)
Comment (AntonTimiskov) [reply http_calls.py:259]: Not critical

File: http_calls.py (hunk 1 above)
Comment (AntonTimiskov) [reply http_calls.py:269]: please revert

Comment (AleksandrIakhnev) [reply http_calls.py:269]: fixed

File: http_calls.py (hunk 4 above)
Comment (AleksandrIakhnev) [reply http_calls.py:169]: fixed

File: http_calls.py (hunk 3 above)
Comment (AleksandrIakhnev) [reply http_calls.py:102]: restored

**Review by AntonTimiskov — APPROVED** [review]

**Pull request merged**